### PR TITLE
perf: reduce symlink syscalls on windows

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

File: src/file_system.rs [hunk 1]
```diff
@@ -233,31 +190,96 @@ fn metadata() {
...
+            {
+                node_compatible_raw_canonicalize(fs::read_link(&path_buf)?)
+            }
+            #[cfg(not(windows))]
+            {
+                fs::read_link(&path_buf)?
+            }
+        };
+        path_buf.pop();
+        if fs::symlink_metadata(&path_buf)?.is_symlink() {
```
Comment (sapphi-red) [inline src/file_system.rs:211]: It's not needed for this PR, but writing down an idea. I guess we can reduce syscalls further more by caching `fs::symlink_metadata`.

File: src/file_system.rs [hunk 2]
```diff
@@ -233,31 +190,96 @@ fn metadata() {
...
+                .map(PathBuf::from)
+                .unwrap_or(canonicalized),
+            Prefix::VerbatimDisk(_) => canonicalized
+                .to_str()
+                .and_then(|s| s.get(LONG_PATH_PREFIX.len()..))
+                .map(PathBuf::from)
+                .unwrap_or(canonicalized),
+            _ => canonicalized,
+        },
+        _ => canonicalized,
```
Comment (sapphi-red) [inline src/file_system.rs]: I guess we should use `strip_prefix` for `LONG_PATH_PREFIX` as well and remove this part.

**Comment by sapphi-red** [comment]
Nice 😀

File: src/file_system.rs (hunk 1 above)
Comment (Brooooooklyn) [reply src/file_system.rs:211]: Let's do it later 😁

**Review by sapphi-red — APPROVED** [review]
💚

**Pull request merged**